interfaces/repo_test.go

```diff
@@ -307,6 +318,21 @@ func (s *RepositorySuite) TestAddPlugFailsWithUnknownInterface(c *C) {
 	c.Assert(s.emptyRepo.AllPlugs(""), HasLen, 0)
 }
 
+func (s *RepositorySuite) TestAddPlugParallelInstance(c *C) {
+	c.Assert(s.testRepo.AllPlugs(""), HasLen, 0)
+	err := s.testRepo.AddPlug(s.plug)
+	c.Assert(err, IsNil)
+	consumer := snaptest.MockInfo(c, consumerYaml, nil)
```

Comment:
I think I would find it slightly clearer when `c.Assert(s.testRepo.AllPlugs(""), HasLen, 1)` was added here. It just feels more natural to me to not jump from zero to two.

I.e.:
```
	c.Assert(s.testRepo.AllPlugs(""), HasLen, 0)
	err := s.testRepo.AddPlug(s.plug)
	c.Assert(err, IsNil)
	c.Assert(s.testRepo.AllPlugs(""), HasLen, 1)

	consumer := snaptest.MockInfo(c, consumerYaml, nil)
	consumer.InstanceKey = "instance"
	err = s.testRepo.AddPlug(consumer.Plugs["plug"])
	c.Assert(err, IsNil)
	c.Assert(s.testRepo.AllPlugs(""), HasLen, 2)
        c.Assert(s.testRepo.Plug(s.plug.Snap.InstanceName(), s.plug.Name), DeepEquals, s.plug)
	c.Assert(s.testRepo.Plug(consumer.InstanceName(), "plug"), DeepEquals, consumer.Plugs["plug"])
```